### ISSUE TITLE
Dynamic Open Graph images

### ISSUE THREAD

**Issue by kelvinlockwood** [Charity-Shop-Exchange/Charity-Shop-Exchange#28]
This closes the loop with sharing and regional subdomains...

When the subdomain url is shared it will display its custom Open Graph image, the image can include the names of the charities that are associated with that county.

https://zeit.co/blog/social-og-image-cards-as-a-service

**Comment by globalmatt** [comment]
@kelvinlockwood Had a bit of a look at this. It seems that it's designed to deploy on ZEIT Now. Not sure how easy it would be to build and deploy it on Netlify. I think Netlify only supports [AWS Lambda functions](https://docs.netlify.com/functions/overview/) whereas Now has [its own APIs for serverless functions](https://zeit.co/docs/v2/serverless-functions/introduction). But this is is all new territory for me :)

I guess we could deploy it on ZEIT Now (which has a [free tier](https://zeit.co/pricing)) and call it from the main site, still running on Netlify?

@Prouty89 any thoughts?

I did also find this - would it do the job? https://www.pictura.io

**Comment by globalmatt** [comment]
OK, I'm going to have a play with the ZEIT thing in a separate repo and see if I can get the dynamic `og:image` stuff working. Then we can decide if we want to use it :)

**Comment by hmajid2301** [comment]
So quick update on this issue. Even assuming we had a design we could dynamically generate using the `zeit og-image tool`. There is one other issue we would need to resolve;

How can we change the `og:image` meta tag dynamically. It seems this cannot be done using JS. If you use http://www.opengraphtester.com/test/ on the https://charityshopexchange.com. You will see the current open graph image as expected. However I have spun up a dummy website which dynamically injects the `og:image` here https://objective-haibt-018286.netlify.app/. If you test this in the same open graph testing tool you will notice there is no image.

Doing some further research [here](https://stackoverflow.com/questions/34281862/dynamic-facebook-open-graph-tags-possible) and [here](https://stackoverflow.com/questions/50193960/react-helmet-facebook-open-graph). Facebook and other such tools dont parse/run the JS so the meta tag is never dynamically created. So they don't have any image to follow.

The solution to this problem would be to server-side render (SSR) the website first. Some examples of tools that do this include next.js and gatsby. You can also deploy this onto Netlify. I am 100% of all the changes required. But I briefly used `next.js` in the past I don't think it's a very large change.

Thoughts @globalmatt and @kelvinlockwood.

**Comment by globalmatt** [comment]
@hmajid2301 Yes, I reached the same conclusion when I researched this last month. We will need to do this server-side using Gatsby or similar I think.
